### ISSUE TITLE
EXTRACT_ALL with GENERATE_TAGFILE creates duplicate entries for \anchor-s

### ISSUE THREAD

**Issue by svarshavchik** [doxygen/doxygen#7711]
I end up with duplicate references to \anchor-s in generated tagfiles when I use EXTRACT_ALL.

Referencing the generated tagfile produces a bunch of spam, like:

/tmp/doxybug/doctags:37: warning: Duplicate anchor anchor_in_description found

Attaching a small tarball with a reproducer for doxygen 1.8.15. Referencing the resulting doctags file from another Doxyfile will result in the above spam.

[doxybug.tar.gz](https://github.com/doxygen/doxygen/files/4500088/doxybug.tar.gz)

**Comment by albert-github** [comment]
Unfortunately in the example the tag file using project was missing, but fortunately it was easy to create, so a complete example: [example.tar.gz](https://github.com/doxygen/doxygen/files/4502273/example.tar.gz)

When running doxygen in the directory p1 a tag file doctags is created and when afterwards running doxygen in p2 we get the warning (problem is still present in the current doxygen version 1.8.18):
```
.../doctags:37: warning: Duplicate anchor anchor_in_description found
```

When looking in the `doctags` tag file we see at line 21 (for the compound "file") and line 33 (for compound "namespace") the line:
```
<docanchor file="namespacemine.html">anchor_in_description</docanchor>
```
both times it is in the:
```
<member kind="typedef">
```
and referring to the same entity.

@doxygen 
- what would be the best approach here?

**Comment by albert-github** [comment]
The problem with the 1.8.18 version is as indicated.
In the current doxygen version (1.10.0) the problem is gone, the part about `typedef`  is only available in the tag file in the `namespace` part and not in the `file` part anymore.
(Problem looks like to be solved in the 1.9.8 version)

I think the problem has been solved in the mean time.
